src/components/searchbar/SearchBar.component.tsx

```diff
@@ -1,10 +1,130 @@
+import React, { useEffect, useRef, useState } from 'react';
+
+import { ArrowDownIcon } from '@/assets/ArrowDown.tsx';
+import { ArrowUpIcon } from '@/assets/ArrowUp.tsx';
+import { filters } from '@/interfaces/filters.ts';
+import type { Product } from '@/interfaces/Product.ts';
+
 import styles from './SearchBar.module.css';
 
-export default function SearchBar() {
+interface SearchProps {
+    products: Product[];
+    currentPage: number;
+    onPageChange: (page: number) => void;
+    onFilteredProducts: (products: Product[]) => void;
+}
+
+export const SearchBar: React.FC<SearchProps> = ({ products, currentPage, onPageChange, onFilteredProducts }) => {
+    const [isDropdownOpen, setIsDropdownOpen] = useState<boolean>(false);
+    const [searchTerm, setSearchTerm] = useState<string>('');
+    const [activeCategories, setActiveCategories] = useState<string[]>([]);
+    const [activeFilter, setActiveFilter] = useState<string>('');
+    const [currentSort, setCurrentSort] = useState<string>('Price (High - Low)');
+    const buttonReference = useRef<HTMLButtonElement>(null);
+    const dropdownReference = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        let filtered = products;
+
+        if (searchTerm) {
+            filtered = filtered.filter((product) => product.title.toLowerCase().includes(searchTerm.toLowerCase()));
+        }
+
+        if (activeCategories.length > 0) {
+            filtered = filtered.filter((product) => activeCategories.includes(product.category.name));
+        }
+
+        if (activeFilter) {
+            switch (activeFilter) {
+                case 'price-asc': {
+                    filtered = filtered.sort((a, b) => a.price - b.price);
+                    break;
+                }
+                case 'price-desc': {
+                    filtered = filtered.sort((a, b) => b.price - a.price);
+                    break;
+                }
+                case 'newest': {
+                    filtered = filtered.sort((a, b) => new Date(b.creationAt).getTime() - new Date(a.creationAt).getTime());
+                    break;
+                }
+                case 'oldest': {
+                    filtered = filtered.sort((a, b) => new Date(a.creationAt).getTime() - new Date(b.creationAt).getTime());
+                    break;
+                }
+            }
+        }
+        onFilteredProducts(filtered);
+    }, [activeFilter, searchTerm, activeCategories, products, currentPage]);
```

Comment:
quick fix:
```suggestion
    }, [activeFilter, searchTerm, activeCategories, products, currentPage, onFilteredProducts]);
```
Linter said: `onFilteredProducts` is a variable (function in this case) you are using inside the `useEffect` hook. So you should add this variable to the dependencies of the hook.